src/components/markdown/Spoiler.tsx

```diff
@@ -1,6 +1,47 @@
-import * as React from 'react';
+import React, { Children } from 'react';
+
+function determineIfSingularCodeBlock(
+  firstName: string,
+  numChildren: number
+): boolean {
+  return firstName == 'LanguageSection' && numChildren == 1;
+}
 
 const Spoiler = ({ children, title }) => {
+  let count = 0;
+  let numChildren = 0;
+  let firstName = 'None';
+  let ogProps;
+
+  React.Children.forEach(children, child => {
+    if (count == 0) {
+      numChildren = React.Children.count(children);
+      firstName = child.props.mdxType;
+      ogProps = child.props;
+    }
+    count++;
+  });
```

Comment:
you shouldn't (?) need a forEach for this? can't you do this without the loop?

Reply:
@thecodingwizard Not really sure how to get the name of the language section without the for-each?

Reply:
https://stackoverflow.com/questions/50870053/pass-props-to-the-first-child-using-this-props-children

https://reactjs.org/docs/jsx-in-depth.html#children-in-jsx

(iirc children is only an array if there's more than one child? not sure)

Reply:
I believe it is always an array, thanks!